loader/loader.go

```diff
@@ -1307,6 +1307,11 @@ func (l *Loader) getMydumpMetadata() error {
 	metafile := filepath.Join(l.cfg.LoaderConfig.Dir, "metadata")
 	loc, _, err := dumpling.ParseMetaData(metafile, l.cfg.Flavor)
 	if err != nil {
+		if terror.ErrMetadataNoBinlogLoc.Equal(err) {
+			l.logger.Warn("dumped metadata doesn't have binlog location, it's OK if DM doesn't enter incremental mode")
```

Comment:
How about check task_mode here? 
It is annoying to report errors after a long time import is completed

Reply:
We're checking REPLICATION CLIENT in pre-check wrongly, (should make sure it's grants on `*.*`). So for all mode and incremental mode, pre-check could make sure there's enough privilege generating correct metadata. For full mode, it's no harm

Reply:
Are there any other cases that may cause this error? 🤔